### PR TITLE
feat(pie-docs): DSW-1514 add divider docs

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)
- adds the overview and code sections of `pie-divider`

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/components/divider/code/code.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+></iframe>
+
+## Variants
+
+ <iframe
+  src="https://webc.pie.design/?path=/docs/divider--variants&viewMode=story&shortcuts=true&singleStory=true"
+  width="100%"
+  height="600px"
+  style="border: none; margin-top: 32px;"
+></iframe>
```
Comment (fernandofranca) [inline apps/pie-docs/src/components/divider/code/code.md:59]: I got this error here:
image

**Review by fernandofranca — DISMISSED** [review]

File: apps/pie-docs/src/components/divider/overview/overview.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+---
+
+## Dos and Don’ts
+
+{% usage {
+    do: {
+        type: usageTypes.text,
+        items: [
+            "Use divider to separate related elements into groups or break up dense content.",
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/divider/overview/overview.md:27]:
```suggestion
            "Use a divider to separate related elements into groups or break up dense content.",
```

File: apps/pie-docs/src/components/divider/overview/overview.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            "Use divider when items can’t be separated with open space."
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/divider/overview/overview.md:28]:
```suggestion
            "Use a divider when items can’t be separated with open space."
```

File: apps/pie-docs/src/components/divider/overview/overview.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    key: Overview
+    parent: Divider
+    order: 1
+shouldShowContents: true
+---
+
+## Overview
+The purpose of dividers is to create a clear visual distinction between elements or sections, improving the overall layout and user experience. It helps users understand the structure and hierarchy of the content, making it easier to navigate and understand the interface.
+
+Dividers are commonly used to visually separate different sections, such as between paragraphs of text, in lists or between distinct components or modules. They provide visual clarity, improve readability and contribute to a more organised and aesthetically pleasing user interface.
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/divider/overview/overview.md:12]:
```suggestion
Dividers are commonly used to visually separate different sections, such as between paragraphs of text, in lists, or between distinct components or modules. They provide visual clarity, improve readability, and contribute to a more organised and aesthetically pleasing user interface.
```

**Review by fernandofranca — APPROVED** [review]

**Review by jamieomaguire — APPROVED** [review]

**Pull request merged**